### ISSUE TITLE
MacOS: error installing MarkupSafe in GitHub Actions

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/buildout.coredev#799]
On coredev 6.0. See [this run](https://github.com/plone/buildout.coredev/runs/7001455580?check_suite_focus=true):

```
Getting distribution for 'MarkupSafe==2.1.1'.
While:
  Installing.
  Loading extensions.
  Getting distribution for 'MarkupSafe==2.1.1'.
An internal error occurred due to a bug in either zc.buildout or in a
recipe being used:
Traceback (most recent call last):
  File "/Users/runner/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/zc/buildout/buildout.py", line 2250, in main
    getattr(buildout, command)(args)
  File "/Users/runner/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/zc/buildout/buildout.py", line 696, in install
    self._load_extensions()
  File "/Users/runner/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/zc/buildout/buildout.py", line 1230, in _load_extensions
    zc.buildout.easy_install.install(
  File "/Users/runner/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/zc/buildout/easy_install.py", line 971, in install
    return installer.install(specs, working_set)
  File "/Users/runner/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/zc/buildout/easy_install.py", line 744, in install
    for dist in self._get_dist(req, ws):
  File "/Users/runner/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/zc/buildout/easy_install.py", line 581, in _get_dist
    dists = [_move_to_eggs_dir_and_compile(dist, self._dest)]
  File "/Users/runner/hostedtoolcache/Python/3.10.5/x64/lib/python3.10/site-packages/zc/buildout/easy_install.py", line 1948, in _move_to_eggs_dir_and_compile
    assert newdist is not None  # newloc above is missing our dist?!
AssertionError
Error: Process completed with exit code 1.
```

I do not see the error locally. Mac OS 10.15.7 here. Used Python 3.10.1 from pyenv until this morning. Updated to 3.10.4, threw all eggs away, tried again: still fine.

Is anyone seeing this locally?

**Comment by mauritsvanrees** [comment]
As a workaround, I exclude MacOS on Python 3.10 on gh-actions. See https://github.com/plone/buildout.coredev/pull/800

**Comment by mauritsvanrees** [comment]
Fixed by PR #802.
